### PR TITLE
Add stack integration

### PR DESCRIPTION
**Pull request by meditans**

### CODE REVIEW

**Comment by mvdan** [comment]
Is this common practice? I thought `stack init` working from a cabal setup was good enough for stack users.

**Comment by mvdan** [comment]
Asking especially because this adds new things to take care of, like the `lts` resolver version.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 dist/
+.stack-work/
 *.o
 *.hi
+TAGS
```
Comment (mvdan) [inline .gitignore]: where did this come from?

**Comment by meditans** [comment]
Right, this is a good question. Take a look at [this thread]( https://www.reddit.com/r/haskell/comments/415vne/to_ship_stackyaml_or_not_to/) in which the question was discussed.

In particular I find that [this comment](https://www.reddit.com/r/haskell/comments/415vne/to_ship_stackyaml_or_not_to/cyzvtk4) and its descendants make a point.

However, if you decide against `stack.yaml` inclusion, adding it to `.gitignore` may be a good thing for those who use it.

File: .gitignore (hunk 1 above)
Comment (meditans) [inline .gitignore]: It's a common file generated by setups that provide a `go to definition` facility (in my case it's [`hasktags`](http://hackage.haskell.org/package/hasktags)). Tell me if you prefer that to be removed

Comment (gelisam) [inline .gitignore]: git has a user-wide gitignore which is more appropriate for this kind of file than the project-specific gitignore.

Comment (meditans) [inline .gitignore]: Thanks @gelisam, didn't know that!

Comment (mvdan) [inline .gitignore]: Agreed - let's leave it out

**Comment by mvdan** [comment]
> However, if you decide against stack.yaml inclusion, adding it to .gitignore may be a good thing for those who use it.

I'd have no problem with it if the resolver wasn't stuck at some arbitrary version. The fact that it needs some maintenance (even if it's ok if it's not kept up to date) I'm not very comfortable with. Not saying that what `cabal` does is better, but I'd rather use one method than maintain two different ones at once.

**Comment by mvdan** [comment]
I don't think adding `stack.yaml` to `.gitignore` makes much sense either - it might come across as some sort of hate for Stack. Seems to me like it's just an attempt on our part to help people who don't know how `git add` works. So I wouldn't add it.

**Pull request closed without merging**